swap/swap.go

```diff
@@ -209,12 +213,17 @@ func New(dbPath string, prvkey *ecdsa.PrivateKey, backendURL string, params *Par
 		chainID.Uint64(),
 		params,
 		factory,
+		chain.NewTxQueue(stateStore, "chain", &chain.DefaultTxSchedulerBackend{
+			Backend: backend,
+		}, owner.privateKey),
 	)
 	// start the chequebook
 	if swap.contract, err = swap.StartChequebook(chequebookAddressFlag); err != nil {
 		return nil, err
 	}
 
+	swap.txScheduler.Start()
```

Comment:
Hm, we should make this change when `Desposit` is actually changed. The problem here is that as soon as `Start` is called stuff might happen which triggers notifications which are handled before the chequebook is set here.

Reply:
That being said there is already a problem here as the `CashoutProcessor` is already receiving notifications at this point. The initialisation of this might need changes.

Reply:
What to do with this comment? @ralph-pichler

Reply:
Initialisation has changed now so that notifications are only delivered after the queue has started. The location of the `swap.txScheduler.Start` call remains unchanged for now, we should modify this when `Deposit` is changed.